merino-settings/src/providers.rs

```diff
@@ -215,6 +217,59 @@ impl Default for StealthConfig {
     }
 }
 
+/// Settings for Merino suggestion providers.
+#[derive(Clone, Debug, Serialize, Deserialize)]
+#[serde(transparent)]
+pub struct SuggestionProviderSettings(pub HashMap<String, SuggestionProviderConfig>);
+
+impl SuggestionProviderSettings {
+    /// Load settings for suggestions providers.
+    ///
+    /// The organization of the provider configuration files is identical to the
+    /// top level settings.
+    ///
+    /// Note that settings for suggestion providers cannot be configured via
+    /// environment variables.
+    ///
+    /// # Errors
+    /// If any of the configured values are invalid, or if any of the required
+    /// configuration files are missing.
+    pub fn load() -> Result<Self> {
+        let mut s = Config::new();
+
+        // Start off with the base config.
+        s.merge(File::with_name("./config/providers/base"))
+            .context("loading base config for suggestion providers")?;
+
+        // Merge in an environment specific config.
+        let merino_env = std::env::var("MERINO_ENV").unwrap_or_else(|_| "development".to_string());
+        s.merge(File::with_name(&format!("./config/providers/{}", merino_env)).required(false))
+            .context("loading environment config for suggestion providers")?;
+
+        // Add a local configuration file that is `.gitignore`ed.
+        s.merge(File::with_name("./config/providers/local").required(false))
+            .context("loading local config for suggestion providers")?;
+
+        serde_path_to_error::deserialize(s)
+            .context("Deserializing settings for suggestion providers")
+    }
+
+    /// Load settings for suggestion providers from configuration files for tests.
+    pub fn load_for_tests() -> Self {
+        let mut s = Config::new();
+
+        s.merge(File::with_name("../config/providers/test"))
```

Comment:
Though there's nothing in there currently I think this should also load `../config/providers/base` first similar to what `Settings` does.

Reply:
Or maybe this doesn't matter since this is solely a `HashMap`? I'm not sure how merge/try_into work here..

Reply:
Testing locally the merging does support this, e.g. a `test_remote_settings` in `test.yaml` and `test_wiki_fruit` in `local_test.yaml` results in both keys included. So it makes sense to also load from `base`.

Reply:
Yeah, since it's backed by `HashMap`, `merge` will override the old one if there is a new one with the same provider ID.

I think the reason I didn't load from `base` is that some special providers (WikiFruit, DebugProvider, and FixedProvider) are only valid under the debug mode, trying to load them for other modes will fail.

I wonder if we should just get rid of `base` because, unlike the top-level settings, I don't see a common provider that's shared across all the different modes. So for suggestion providers, we will have a scenario-specific configuration (e.g. dev, prod, test, ci, etc), a local one (optional & gitignored), and another remote one for production.

@pjenvey @mythmon What do you all think?

Reply:
I think you're right that there aren't currently any common providers we'd want to share across providers. Maybe that will change in the future, but if it does then we can add this back. Also as I explained elsewhere, I think we should only support a single configuration source for this anyways.

Reply:
Having a single source and loading the source path/URL from the top-level settings sounds like a good idea to me. As for those local sources (`providers/local.yaml` & `providers/local_test.yaml`), I think folks could still use them by overriding the source path in their `local.yaml`. Then all the major scenarios should be covered by this.

Reply:
Works for me